src/components/ArticleCard.jsx

```diff
@@ -9,16 +9,33 @@ function ArticleCard() {
     const {article_id}= useParams()
     const [article,setArticle]=useState([])
     const[comments,setComments]=useState([])
+    const [clicked,setClicked]=useState(0)
+    const [error,setError]=useState()
     useEffect(()=>{
         getArticleById(article_id).then((article)=>{
             setArticle(article)
         }
         )
     },[])
+   
+    function changeVotes({article_id},vote) {
+        if(clicked===0){
+            setArticle((currArticle)=>{
+                setClicked(1)
+                return{...currArticle,votes:article.votes + 1}
+            })    
+        incrementArticleVoteCount(article_id,vote)
```

Comment:
Nice optimistic rendering. You may want a catch block that un-does the increase if anything goes wrong 👍